### PR TITLE
Separate `schemes` in `apinetlet` into `managerScheme` and `apinetletScheme`

### LINKED ISSUE #309: Separate `schemes` in `apinetlet` into `managerScheme` and `apinetletScheme`
**Describe the bug**

Separate `schemes` in the `apinetlet` into a `managerScheme` and an `apinetletScheme`. Use those 2 scheme in the corresponding clients instead of relying on the same. This will cause a problem when you separate the `ironcore` from the `ironcore-net` cluster.

### PR DESCRIPTION
**Pull request by balpert89**
Fixes #309

### CODE REVIEW

**Comment by balpert89** [comment]
@sujeet01 @Rohit-0505 https://github.com/ironcore-dev/ironcore-net/pull/310/files#diff-cd513991cc10b556eb34e9aa31ffbe2e3090219f8ff47f0688ea2e07250a2db3R665-R673 these enqueue funcs produce the following error:
`2024-07-29T12:45:28Z    ERROR   controller-runtime.source.EventHandler  kind must be registered to the Scheme   {"error": "no kind is registered for the type v1alpha1.Network in scheme \"pkg/runtime/scheme.go:100\""}`

The other controllers are running as expected, please have a look and fix.

@afritzler hope the styling is okay, my formatter is a bit more picky :)

File: apinetlet/controllers/network_controller.go [hunk 1]
```diff
@@ -206,15 +245,23 @@ func (r *NetworkReconciler) applyAPINetNetwork(ctx context.Context, log logr.Log
...
+				if peeringNetworkNamespace == "" {
+					peeringNetworkNamespace = network.Namespace
+				}
+				return peering.NetworkRef.Name == peeringClaimRef.Name &&
+					peeringNetworkNamespace == peeringClaimRef.Namespace
+			},
+		)
 		if idx != -1 {
-			peeringPrefixes, err := r.getAPINetNetworkPeeringPrefixes(ctx, network.Spec.Peerings[idx].Prefixes, network.Namespace)
+			peeringPrefixes, err := r.getAPINetNetworkPeeringPrefixes(
```
Comment (afritzler) [inline apinetlet/controllers/network_controller.go:260]: Single line please.

File: apinetlet/controllers/network_controller.go [hunk 2]
```diff
@@ -279,7 +329,7 @@ func (r *NetworkReconciler) SetupWithManager(mgr ctrl.Manager, apiNetCache cache
 		).
 		WatchesRawSource(
 			source.Kind(apiNetCache, &apinetv1alpha1.Network{}),
-			handler.EnqueueRequestForSource(mgr.GetScheme(), mgr.GetRESTMapper(), &networkingv1alpha1.Network{}),
+			handler.EnqueueRequestForSource(r.APINetClient.Scheme(), r.APINetClient.RESTMapper(), &networkingv1alpha1.Network{}),
```
Comment (afritzler) [inline apinetlet/controllers/network_controller.go:332]: We have to use the mgr client here.

**Review by afritzler — CHANGES_REQUESTED** [review]
Thanks @balpert89 for you PR. Can we please keep the method signature and invocation in a single line as we have this code style everywhere else. Only when the parameters are going over 100 chars we typically put them on individual lines.

**Comment by balpert89** [comment]
@sujeet01 will continue from here

**Comment by balpert89** [comment]
closing this one as agreed we do not need the scheme changes and so on.

**Pull request closed without merging**